openmc/tallies.py

```diff
@@ -1453,6 +1476,33 @@ def _outer_product(self, other, binary_op):
 
         """
 
+        # Set default value for filter product if it was not set
+        if filter_product == 'None':
+            filter_product = 'entrywise'
+        elif filter_product == 'tensor':
+            msg = 'Unable to perform Tally arithmetic with a tensor product' \
+                  'for the filter data as this not currently supported.'
+            raise ValueError(msg)
+
+        # Set default value for nuclide product if it was not set
+        if nuclide_product == 'None':
+            if self.nuclides == other.nuclides:
+                nuclide_product = 'entrywise'
+            else:
+                nuclide_product = 'tensor'
+
+        # Set default value for score product if it was not set
+        if score_product == 'None':
+            if self.scores == other.scores:
+                score_product = 'entrywise'
+            else:
+                score_product = 'tensor'
+
+        # Check product types
+        cv.check_value('filter product', filter_product, _PRODUCT_TYPES)
```

Comment:
The lines above set the products if they were not input, so they should be either `entrywise` or `tensor` at this point.